src/main/java/seedu/address/logic/parser/ViewCommandParser.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_CLASS;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ViewCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ViewCommand object
+ */
+public class ViewCommandParser implements Parser<ViewCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the ViewCommand
+     * and returns an ViewCommand object for execution.
+     * @throws ParseException if the user input does not follow the expected format
+     */
+    public ViewCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CLASS);
+
+        // ensures that class index is provided
+        if (!arePrefixesPresent(argMultimap, PREFIX_CLASS)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_USAGE));
+        }
+
+        //ensures that the index provided is more than 0
+        if (Integer.parseInt(argMultimap.getValue(PREFIX_CLASS).get()) <= 0) {
```

Comment:
Nice documentation of your code, really helps me understand it with a glance. 👍 👍 
But just a question, will the case of index greater than size of filtered list be accounted for?
Please ignore if I'm wrong, I don't really remember if it has been encapsulated in parseIndex